src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -25,23 +27,32 @@
     /**
      * Ignore the styles from the generic folder.
      */
-    private static final String NULL_STYLE = "schematics";
+    private static final String NULL_STYLE = "";
 
     /**
      * Ignore the styles from the miner folder.
      */
     private static final String MINER_STYLE = "miner";
 
-    private static final String                    SCHEMATIC_EXTENSION   = ".nbt";
-    private static final String                    SCHEMATICS_ASSET_PATH = "/assets/minecolonies/schematics/";
+    private static final String                    SCHEMATIC_EXTENSION        = ".nbt";
+    private static final String                    SCHEMATICS_ASSET_PATH      = "/assets/minecolonies/schematics/";
+    public  static final String                    SCHEMATICS_HUTS            = "huts";
+    public  static final String                    SCHEMATICS_DECORATIONS     = "decorations";
     //Hut, Levels
     @NotNull
-    private static final Map<String, Integer>      hutLevelsMap          = new HashMap<>();
+    private static       Map<String, Integer>      hutLevelsMap          = new HashMap<>();
     //Hut, Styles
     private static       Map<String, List<String>> hutStyleMap           = new HashMap<>();
     //Decoration, Style
     private static       Map<String, List<String>> decorationStyleMap    = new HashMap<>();
 
+    /* md5 hash for the schematics
+     * format is:
+     * huts/stone/builder1.nbt -> hash
+     * decorations/decoration/Well.nbt -> hash
+     */
+    private static       Map<String, String>       md5Map                = new HashMap<>();
```

Comment:
The only reason why I had duplicated schematic is because I was testing the custom huts features and see if the clients pick up the new schematics.

I decided to go for the TowerGuard because I was in need to have a corner version.

I asked my builder to build the 5 different normal TowerGuards, one for each level.

Modified them to do the corner version. then scan them.

I did not rescan the level 1 because I did not change anything, and just copy the TowerGuard1 to the **Corner** style.

Also later one, player will be be able to scan. My plan is to store those in "schematics/players/${player UUID}/"
I can not control what player will scan if two players scan a 2x2x2 Cobblestone schematic, they will have the same MD5 hash. but each schematics will be in their own  player directory, so that each player can only uses their schematic not what another player have done.